libs/datamodel/core/src/transform/ast_to_dml/db.rs

```diff
@@ -85,8 +85,9 @@ impl<'ast> ParserDatabase<'ast> {
                         )),
                     };
                 }
-                ast::Top::Model(model) => visit_model(top_id, model, &mut ctx),
-                ast::Top::Source(_) | ast::Top::Generator(_) | ast::Top::Enum(_) => (),
+                (ast::TopId::Model(model_id), ast::Top::Model(model)) => visit_model(model_id, model, &mut ctx),
+                (_, ast::Top::Source(_)) | (_, ast::Top::Generator(_)) | (_, ast::Top::Enum(_)) => (),
+                _ => unreachable!(),
```

Comment:
These are temporary. Once we don't use `TopId` anymore (it's still used in names.rs), we can return something like `enum TopItem { Model(ModelId, &ast::Model), ...}` from `SchemaAst::iter_tops()`.